src/types/type-declarations.ts

```diff
@@ -100,16 +110,15 @@ type GoogleTagManagerDataLayerStatus = {
   };
 };
 
-export type GoogleTagManager = GoogleTagManagerDataLayerStatus & {
+export type GoogleTagManagerInstance = GoogleTagManagerDataLayerStatus & {
   [key: string]: {
     callback: () => void;
     dataLayer: GoogleTagManagerDataLayerApi;
   };
 };
 
 export interface GoogleTagManagerApi {
-  dataLayers: Record<string, Record<string, any>[]>;
-  google_tag_manager: GoogleTagManager;
+  [key: string]: Maybe<DataLayer> | Maybe<GoogleTagManagerInstance>;
```

Comment:
google_tag_manager doesn't need to be removed, nor being wrapper with Maybe i think